invertedai/logs/common.py

```diff
@@ -0,0 +1,456 @@
+                        "speed": 0,
+                        "control_state": tls[actor_id]
+                    }
+
+                predetermined_controls_dict[actor_id] = {
+                    "entity_type": "traffic_light",
+                    "static_attributes": {
+                        "length": actor.length,
+                        "width": actor.width,
+                        "rear_axis_offset": 0,
+                    },
+                    "states":states_dict
+                }
+
+        individual_suggestions_dict = {}
+        for i, prop in enumerate(scenario_log.agent_properties):
+            wp = prop.waypoint
+            if wp is not None:
+                individual_suggestions_dict[str(i)] = {
+                    "suggestion_strength": 0.8,
+                    "states": {
+                        "0": {
+                            "center": {
+                                "x": wp.center.x,
+                                "y": wp.center.y
+                            }
+                        }
+                    }
+                }
+
+        output_dict = {
+            "location": {
+                "identifier": scenario_log.location
+            },
+            "scenario_length": len(scenario_log.agent_states),
+            "num_agents": {
+                "car": num_cars,
+                "pedestrian": num_pedestrians
+            },
+            "predetermined_agents": predetermined_agents_dict,
+            "num_controls": {
+                "traffic_light": num_controls_light,
+                "yield_sign": num_controls_yield,
+                "stop_sign": num_controls_stop,
+                "other": num_controls_other,
+            },
+            "predetermined_controls": predetermined_controls_dict,
+            "individual_suggestions": individual_suggestions_dict,
+            "initialize_random_seed": scenario_log.initialize_random_seed,
+            "lights_random_seed": scenario_log.lights_random_seed,
+            "drive_random_seed": scenario_log.drive_random_seed,
+            "drive_model_version": scenario_log.drive_model_version,
+            "initialize_model_version": scenario_log.initialize_model_version,
+            "birdview_options": {
+                "rendering_center": [
+                    scenario_log.rendering_center[0],
+                    scenario_log.rendering_center[1]
+                ],
+                "renderingFOV": scenario_log.rendering_fov
+            },
+            "light_recurrent_states": [] if scenario_log.light_recurrent_states is None else [lrs.tolist() for lrs in scenario_log.light_recurrent_states],
+            "rendering_centers": [
+                scenario_log.rendering_center[0],
+                scenario_log.rendering_center[1]
+            ]
+        }
+
+        with open(log_path, "w") as outfile:
+            json.dump(output_dict, outfile)
+
+    @classmethod
+    def write_json_from_scenario_log(
```

Comment:
This seems to be doing the same thing as `write_scenario_log_to_json` ?  And it doesn't make any sense to pass a cls to an instance method. I am not sure if this will work without passing a concrete object to it

Reply:
Tested this and it indeed works. I can see it being useful if a user doesn't want to create a LogWriter object just to write the log to disk.